zkvm/src/signature/multikey.rs

```diff
@@ -1,17 +1,29 @@
 use super::VerificationKey;
+use crate::errors::VMError;
 use crate::transcript::TranscriptProtocol;
 use curve25519_dalek::ristretto::RistrettoPoint;
 use curve25519_dalek::scalar::Scalar;
 use merlin::Transcript;
 
 #[derive(Clone)]
 pub struct Multikey {
-    transcript: Transcript,
+    transcript: Option<Transcript>,
     aggregated_key: VerificationKey,
 }
 
 impl Multikey {
-    pub fn new(pubkeys: Vec<VerificationKey>) -> Option<Self> {
+    pub fn new(pubkeys: Vec<VerificationKey>) -> Result<Self, VMError> {
```

Comment:
if we are going with Result<> i'd prefer not tying MuSig things into VMError/ZkVM too much. Maybe not in this PR, but it'd be cool to have MuSigError type, and keep all the things in one module. And then maybe move it out into a separate crate. I'll file an issue.

Reply:
sgtm